### PR TITLE
Some things I didn't catch about the network types/timers code

### PR DESCRIPTION
**Pull request by evan-masseau**
# Purpose

**Type of PR**
* [ ] Bug Fix
* [x] Feature Work
* [ ] Revert
* [ ] Refactor / Code Cleanup
* [ ] Other (fill in...)

## Short Description

I just noticed some more things about the network timers PR jordan and I merged yesterday once i actually started looking at the code in android studio.

## Changelog / Code Overview

Fixed a logical hole where transport types other than wifi or cell would be treated as offline 
Added Test coverage
Some other trivial things like enum casing and using 1 config method with an argument instead of 3 separate methods.

## Test Plan

Added coverage

### CODE REVIEW

**Comment by evan-masseau** [comment]
@jordan-griffin i'll tag you in case you want to see this later, but obvs not worth interrupting your PTO.

File: sdk/analytics/src/test/java/com/klaviyo/analytics/networking/KlaviyoApiClientTest.kt [hunk 1]
```diff
@@ -84,7 +87,20 @@ internal class KlaviyoApiClientTest : BaseTest() {
         mockk<KlaviyoApiRequest>().also {
             every { it.uuid } returns uuid
             every { it.send() } returns status
-            every { it.toJson() } returns "{\"headers\":{\"headerKey\":\"headerValue\"},\"method\":\"GET\",\"query\":{\"queryKey\":\"queryValue\"},\"time\":\"time\",\"uuid\":\"$uuid\",\"url_path\":\"test\"}"
+            every { it.toJson() } returns """
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/networking/KlaviyoApiClientTest.kt:90]: this is unrelated just something i've been changing when i see it in tests, much easier to read multiline strings

File: sdk/core/src/main/java/com/klaviyo/core/networking/NetworkMonitor.kt [hunk 2]
```diff
@@ -33,21 +33,15 @@ interface NetworkMonitor {
...
     /**
      * Enum class representing the different network connection types that may affect how our
      * SDK operates.
      */
     enum class NetworkType(val position: Int) {
-        WIFI(0),
-        CELL(1),
-        OFFLINE(2);
-
-        companion object {
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/networking/NetworkMonitor.kt:47]: remove unused

File: sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt [hunk 3]
```diff
@@ -57,15 +56,20 @@ internal object KlaviyoNetworkMonitor : NetworkMonitor {
...
-        if (connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true) {
-            return NetworkMonitor.NetworkType.WIFI.position
-        } else if (connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true) {
-            return NetworkMonitor.NetworkType.CELL.position
+    override fun getNetworkType(): NetworkMonitor.NetworkType {
+        val net = connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+
+        return if (net?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true) {
+            NetworkMonitor.NetworkType.Wifi
+        } else if (net?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true ||
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt:66]: Technically internet can come from a few other transport methods like bluetooth, so this will just catch those and treat them as cellular (maybe we should call our types something more generic like Stable and Unstable idk)

File: sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt [hunk 4]
```diff
@@ -120,25 +133,12 @@ object KlaviyoConfig : Config {
...
-        override fun networkFlushIntervalCell(networkFlushInterval: Int) = apply {
-            if (networkFlushInterval >= 0) {
-                this.networkFlushIntervals[NetworkMonitor.NetworkType.CELL.position] = networkFlushInterval
-            } else {
-                // TODO Logging
-            }
-        }
-
-        override fun networkFlushIntervalOffline(networkFlushInterval: Int) = apply {
+        override fun networkFlushInterval(
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:136]: Use a single method that takes type as an argument instead of 3 methods

File: sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt [hunk 5]
```diff
@@ -169,11 +169,14 @@ object KlaviyoConfig : Config {
...
 
             val permissions = applicationContext!!.packageManager.getPackageInfoCompat(
-                applicationContext!!.packageName, PackageManager.GET_PERMISSIONS
+                applicationContext!!.packageName,
+                PackageManager.GET_PERMISSIONS
             ).requestedPermissions ?: emptyArray()
 
-            if (Manifest.permission.ACCESS_NETWORK_STATE !in permissions) {
-                throw MissingPermission(Manifest.permission.ACCESS_NETWORK_STATE)
+            requiredPermissions.forEach { permission ->
```
Comment (evan-masseau) [inline sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:176]: semi-related, we require two different network permissions, so this checks both.

File: sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt (hunk 3 above)
Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/core/networking/KlaviyoNetworkMonitor.kt:66]: Neat!

File: sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt (hunk 5 above)
Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:176]: Is caught by anything in the sdk or it will crash the app potentially?

Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:176]: I think that's ok since we can't do much without internet.

Comment (evan-masseau) [reply sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:176]: Any of these config errors we throw will crash the app if uncaught, yes. Its either that or the system crashes us later when we try to access those things that are gated by this permission. 
However, we actually declare these permissions in our core sdk manifest. I'm not sure if there is some way to exclude our manifest or something, so I put this in for extra safety

Comment (ndurell) [reply sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt:176]: Makes sense.

**Comment by evan-masseau** [comment]
Was merged with the logging branch
